Hi,

On Wed, 18 Aug 2004, Finn Thain wrote:

> I'm afraid it doesn't boot for me. Penguin hangs after logging "killing
> sonic ethernet", and before logging "booting linux, fasten seatbelts
> please".

The patch below should work better. The second program header has to start 
after the bss.

bye, Roman

Index: arch/m68k/kernel/vmlinux-std.lds
===================================================================
RCS file: /home/linux-m68k/cvsroot/linux/arch/m68k/kernel/vmlinux-std.lds,v
retrieving revision 1.3
diff -u -p -r1.3 vmlinux-std.lds
--- arch/m68k/kernel/vmlinux-std.lds	10 May 2004 18:39:29 -0000	1.3
+++ arch/m68k/kernel/vmlinux-std.lds	19 Aug 2004 00:31:26 -0000
@@ -15,7 +15,7 @@ SECTIONS
 	SCHED_TEXT
 	*(.fixup)
 	*(.gnu.warning)
-	} = 0x4e75
+	} :text = 0x4e75
 
   . = ALIGN(16);		/* Exception table */
   __start___ex_table = .;
@@ -34,7 +34,7 @@ SECTIONS
   .bss : { *(.bss) }		/* BSS */
 
   . = ALIGN(16);
-  .data.cacheline_aligned : { *(.data.cacheline_aligned) }
+  .data.cacheline_aligned : { *(.data.cacheline_aligned) } :data
 
   _edata = .;			/* End of data section */
 
Index: arch/m68k/kernel/vmlinux.lds.S
===================================================================
RCS file: /home/linux-m68k/cvsroot/linux/arch/m68k/kernel/vmlinux.lds.S,v
retrieving revision 1.1.1.1
diff -u -p -r1.1.1.1 vmlinux.lds.S
--- arch/m68k/kernel/vmlinux.lds.S	23 Aug 2003 12:05:42 -0000	1.1.1.1
+++ arch/m68k/kernel/vmlinux.lds.S	19 Aug 2004 00:31:26 -0000
@@ -1,5 +1,9 @@
 #include <linux/config.h>
-
+PHDRS
+{
+  text PT_LOAD FILEHDR PHDRS FLAGS (7);
+  data PT_LOAD FLAGS (7);
+}
 #ifdef CONFIG_SUN3
 #include "vmlinux-sun3.lds"
 #else
